### ISSUE TITLE
--semihosting-cmdline with double-dash option doesn't work

### ISSUE THREAD

**Issue by vjpr** [gnuarmeclipse/qemu#24]
_Command_

```
qemu-system-gnuarmeclipse  \
--gdb tcp::1234 \
-D qemu-log.txt \
-d unimp,guest_errors \
-monitor null \
-serial null \
--verbose \
--verbose \
--board NUCLEO-F411RE \
--mcu STM32F411RE \
--nographic \
--image bin/debug/app.elf \
--semihosting-config enable=on,target=native \
--semihosting-cmdline "--devMode=tag"
```

_Output_

```
GNU ARM Eclipse 64-bits QEMU v2.4.50 (qemu-system-gnuarmeclipse).
qemu-system-gnuarmeclipse: --devMode=tag: invalid option
```

Works:

`--semihosting-cmdline -d tag`

Variation which don't work:

`--semihosting-cmdline --devMode=tag` (no quotes)

It appears that when a `--` or a `=` is used, QEMU tries to parse them.

**Comment by vjpr** [comment]
So my problem was that I had to pass the name of the app (a string starting with no dashes) as the first argument without dashes.

E.g. `--semihosting-cmdline app -d tag`

This was already mentioned in the docs, but was only a recommendation. It seems that it _always_ needs to be the app name.

**Comment by vjpr** [comment]
Also, in the Eclipse UI I have to make the cli command input:

```
"app --devMode=tag "
```

Note the trailing space after tag and before the quote. Without this, the argument is getting parsed as `tag"`.

Also, if I remove the quotes from this input, QEMU attempts to parse them instead.

If I put a quote at the start, then a quote is automatically added to the end which gets included in `argv`.

**Comment by ilg-ul** [comment]
thank you for the report, I'll investigate.

**Comment by ilg-ul** [comment]
fixed in 2.8
